libdistopia/src/distopia.cpp

```diff
@@ -61,12 +62,23 @@ namespace distopia {
         struct OrthogonalBox {
             hn::VFromD<D> lx, ly, lz, ix, iy, iz;
             explicit OrthogonalBox(D d, const T *sbox) {
-                this->lx = hn::Set(d, sbox[0]);
-                this->ly = hn::Set(d, sbox[1]);
-                this->lz = hn::Set(d, sbox[2]);
-                this->ix = hn::Set(d, 1 / sbox[0]);
-                this->iy = hn::Set(d, 1 / sbox[1]);
-                this->iz = hn::Set(d, 1 / sbox[2]);
+                // take norm of x box element
+                T x_vec[3] = {sbox[0], sbox[3], sbox[6]};
+                T x_len = std::sqrt(std::inner_product(x_vec, x_vec + 3, x_vec, 0.0));
+                // take norm of y box element
+                T y_vec[3] = {sbox[1], sbox[4], sbox[7]};
+                T y_len = std::sqrt(std::inner_product(y_vec, y_vec + 3, y_vec, 0.0));
+                // take norm of z box element
+                T z_vec[3] = {sbox[2], sbox[5], sbox[8]};
+                T z_len = std::sqrt(std::inner_product(z_vec, z_vec + 3, z_vec, 0.0));
+                
```

Comment:
@richardjgowers there must be a better way to do this rather than converting to lengths? This many ops is costingus precision and causing stochastic failures (can't reproduce)

Reply:
I'm not sure what you're doing here.  The input to this box should always have 0 for non diagonal elements

Reply:
This is transforming from the box mat, back to the L,L, L, a, b, c repr, but I think after some thinking this is stupid.

Reply:
yeah a=b=c=90 here so it's not worth calculating